src/hidimstat/dcrt.py

```diff
@@ -279,106 +387,261 @@ def _lasso_distillation_residual(
     X,
     y,
     idx,
-    coef_full,
-    Sigma_X=None,
-    cv=3,
-    n_alphas=50,
-    alpha=None,
+    coef_full=None,
+    sigma_X=None,
     n_jobs=1,
-    use_cv=False,
     fit_y=False,
+    random_state=42,
+    params_lasso_distillation_x={
+        "alpha": None,
+        "n_alphas": 10,
+        "alphas": None,
+        "alpha_max_fraction": 0.5,
+    },
+    params_lasso_distillation_y={
+        "alpha": None,
+        "n_alphas": 10,
+        "alphas": None,
+        "alpha_max_fraction": 0.5,
+    },
 ):
     """
-    Standard Lasso Distillation following Liu et al. (2020) section 2.4. Only
-    works for least square loss regression.
-    """
-    n_samples, _ = X.shape
+    Standard Lasso Distillation for least squares regression.
+
+    This function implements the distillation process following :footcite:t:`liu2022fast`
+    section 2.3. It distills both X[:, idx] and y to compute test statistics.
+    It's based on least square loss regression.
 
+    Parameters
+    ----------
+    X : {array-like, sparse matrix} of shape (n_samples, n_features)
+        The input samples.
+    y : array-like of shape (n_samples,)
+        The target values.
+    idx : int
+        Index of the variable to be tested.
+    coef_full : array-like of shape (n_features,), default=None
+        Pre-computed coefficients for y prediction. If None, computed via Lasso.
+    sigma_X : {array-like, sparse matrix} of shape (n_features, n_features), default=None
+        The covariance matrix of X.
+    n_jobs : int, default=1
+        Number of CPUs to use.
+    fit_y : bool, default=False
+        Whether to fit y using Lasso when coef_full is None.
+    random_state : int, default=42
+        Random seed for reproducibility.
+    kargs_lasso_distillation_x : dict
+        Parameters for main Lasso estimation or crossvalidation Lasso, including:
+        - alpha : float, optional - L1 regularization strength. If None, determined by CV.
+        - n_alphas : int, default=0 - Number of alphas for cross-validation.
+        - alphas : array-like, default=None - List of alpha values to try in CV.
+        - alpha_max_fraction : float, default=0.5 - Scale factor for alpha_max.
+    kargs_lasso_distillation_y : dict
+        same as kargs_lasso_distillation_x
+
+    Returns
+    -------
+    ts : float
+        The computed test statistic.
```

Comment:
Do we define what statistic is returned: a correlation ? gaussian variate ? Student ?  p-value ?